modules/miner/mine.go

```diff
@@ -38,48 +50,64 @@ func (m *Miner) blockForWork() (b types.Block) {
 		b.Timestamp = m.earliestTimestamp
 	}
 
-	return
+	return b, b.MerkleRoot(), m.target
+}
+
+// submitBlock takes a solved block and submits it to the blockchain.
+// submitBlock should not be called with a lock.
+func (m *Miner) submitBlock(b types.Block) error {
+	// Give the block to the consensus set.
+	err := m.cs.AcceptBlock(b)
+	if err != nil {
+		m.mu.Lock()
+		fmt.Println("Mined a bad block:", err)
+		fmt.Println(b.ID())
+		childtarget, _ := m.cs.ChildTarget(b.ID())
+		fmt.Println(childtarget)
+		fmt.Println(b.Nonce)
+		m.tpool.PurgeTransactionPool()
+		m.mu.Unlock()
+		return err
+	}
+	if build.Release != "testing" {
+		fmt.Println("Found a block! Reward will be received in 50 blocks.")
+	}
```

Comment:
now that the miner keeps track of the blocks it's mined, we can move the "found block" messages to the UI. (Not in this PR, but eventually.)